packages/webapp/src/components/Forms/ManageCustomTypes/index.jsx

```diff
@@ -0,0 +1,89 @@
+/*
+ *  Copyright 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import { useForm } from 'react-hook-form';
+import { PropTypes } from 'prop-types';
+import { AddLink } from '../../Typography';
+import PageTitle from '../../PageTitle/v2';
+import Form from '../../Form';
+import Tiles from '../../Tile/Tiles';
+import IconLabelTile from '../../Tile/IconLabelTile';
+import { tileTypes } from '../../Tile/constants';
+
+export default function ManageCustomTypes({
+  title,
+  handleGoBack,
+  addLinkText,
+  onAddType,
+  tileData,
+  formatTileData,
+  customTypeFieldName,
+  persistedFormData,
+  useHookFormPersist,
+  getIsSelected,
+  onTileClick,
+}) {
+  const { getValues, setValue, register, watch } = useForm({
+    defaultValues: persistedFormData,
+    mode: 'onChange',
+  });
+
+  useHookFormPersist(getValues);
+  register(customTypeFieldName);
+  const selectedTileId = watch(customTypeFieldName);
+
+  return (
+    <Form>
+      <PageTitle style={{ marginBottom: '20px' }} title={title} onGoBack={handleGoBack} />
+      <AddLink style={{ paddingBottom: '20px' }} onClick={onAddType}>
+        {addLinkText}
+      </AddLink>
+      <Tiles tileType={tileTypes.ICON_LABEL} tileData={tileData} formatTileData={formatTileData}>
+        {tileData.map((data) => {
+          const tileProps = formatTileData(data);
+          const { tileKey } = tileProps;
+          const selected = getIsSelected ? getIsSelected(tileKey) : selectedTileId === tileKey;
+
+          return (
+            <IconLabelTile
+              {...tileProps}
+              key={tileKey}
+              selected={selected}
+              onClick={() => {
```

Comment:
So the way I have the readonly and edit views set up is that the click should just lead to `/readonly_custom_expense/:expense_type_id`. Let me know if I did something wrong there.

[PR-2848](https://github.com/LiteFarmOrg/LiteFarm/pull/2848/files)

Reply:
I tested it and thought it was working perfect!
Actual page navigation is [here](https://github.com/LiteFarmOrg/LiteFarm/pull/2859/files#r1316520904).

Reply:
Sorry yes I saw the other function after the previous comment! It does indeed go where needed. I think I was thrown off by the complexity of the persisted paths. It seemed like it could possibly be a simple prop to pass down. Maybe [here](https://github.com/LiteFarmOrg/LiteFarm/pull/2859/files/3eb45a65ce3fd9eeac4eda6506e1488546318418#r1317323400) is what my brain thought of

Reply:
Ah understood!
I'll do some more experiment with your branch #2848 after the conflicts are resolved. (I will be working on something else, so no need to rush!)